ls1intum/Artemis
# Bugfix/programming exercise/delete exercise

### Description

If the short name of a course was changed at some point in time, programming exercises could not be properly deleted, because the delete methods sometimes used the new project key based on the changed course short name instead of the original repo and build plan URLs.
Besides that, the test repository could not be loaded in the instructor editor, because we also used the wrong URL based on the newer short name in this case.

We now return and use the correct URLs in these cases by only using the original values stored in the participation and exercise entities in the DB.

### Steps for Testing



1. Log in to Artemis
2. Navigate to Course Administration
3. Create a new programming exercise
4. Change the short name of the course of the exercise
5. Open the instructor editor for the exercise and load the test repository
6. The repository status and files should be loaded correctly
7. You can delete the exercise and all projects on the CI and VCS servers will also be deleted
